libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationViewModel.java

```diff
@@ -206,16 +204,17 @@ private void initNavigationLocationEngine() {
     locationEngineConductor = new LocationEngineConductor(locationEngineCallback);
   }
 
-  private void initLanguage(NavigationUiOptions options, LocaleUtils localeUtils) {
-    language = localeUtils.getNonEmptyLanguage(getApplication(), options.directionsRoute().voiceLanguage());
+  private void initLanguage(NavigationUiOptions options) {
+    language = options.directionsRoute().routeOptions().language();
   }
 
-  private void initUnitType(NavigationUiOptions options, LocaleUtils localeUtils) {
+  private void initUnitType(NavigationUiOptions options) {
```

Comment:
What about passing the `unitType` instead of the whole `NavigationUiOptions` object? IMO it's clearer and to the point.

Reply:
@Guardiola31337 see other comment